app/parse.py

```diff
@@ -1,15 +1,59 @@
+import csv
+import requests
+from bs4 import BeautifulSoup
 from dataclasses import dataclass
+from typing import List
+
+
+BASE_URL = "https://quotes.toscrape.com"
 
 
 @dataclass
 class Quote:
     text: str
     author: str
-    tags: list[str]
+    tags: List[str]
+
+
+def parse_single_quote(quote_element) -> Quote:
+    text = quote_element.select_one(".text").get_text()
+    author = quote_element.select_one(".author").get_text()
+    tags = [tag.get_text() for tag in quote_element.select("a.tag")]
+    return Quote(text=text, author=author, tags=tags)
+
+
+def parse_quotes_from_page(page_url: str) -> List[Quote]:
+    response = requests.get(page_url)
+    response.raise_for_status()
+    soup = BeautifulSoup(response.text, "html.parser")
+    quote_elements = soup.find_all("div", class_="quote")
+    return [parse_single_quote(quote_element) for quote_element in quote_elements]
+
+
+def get_all_quotes() -> List[Quote]:
+    all_quotes = []
+    page_url = "/page/1"
```

Comment:
The initial assignment of `page_url` to '/page/1' could be more intuitive if it directly uses the full URL, i.e., `BASE_URL + '/page/1'`. This would make the code more readable and consistent with how URLs are handled later in the function.